lib/bankaccount/account.rb

```diff
@@ -0,0 +1,31 @@
+module BankAccount
+  class Account
+    attr_reader :id, :balance, :initial_balance
+
+    def initialize(id, initial_balance, required_opening_balance = 0)
+      # account cannot be created with negative balance [breakfast]
```

Comment:
Using default arguments is one way to handle not having to override the same method in classes that inherit from a parent class. However, in this case, since the only type of account that has a required opening balance is `SavingsAccount`, making it a concern of the `Account` class might be considered bad practice, since it goes against OO principles. I might suggest using a constant in SavingsAccount that the `initialize` method can check there before calling `super` into the parent method.